VQ2D/README.md

```diff
@@ -45,87 +44,130 @@ Also see the [VQ2D Quickstart Colab Notebook](https://colab.research.google.com/
+
+## Evaluating models on VQ2D
+
+We split the evaluation into two steps: (1) Extracting per-frame bbox proposals and estimating their similarity to the visual query, and (2) Peak detection and bidirectional tracking to infer the response track. There are two key benefits to this separation:
+
+* **Rapid hyperparameter searches for step (2):** Step (1) is the most expensive operation as it takes ~24 hours on an 8-GPU + 80-core machine. Once the detections are pre-computed, step (2) only takes ~1-2 hours on the same machine. This allowed us to release improved hyperparameters for step (2) and obtain much better results.
+* **Decoupling detector model from our inference code for step (2):** While we support only training the baseline model from the [Ego4D paper](https://arxiv.org/pdf/2110.07058.pdf), we can support inference with arbitrary models as long as the pre-extracted detection scores are available.
+
+**Step (1)** Extracting per-frame bbox proposals.
+```
+# Note: MODEL_ROOT and DETECTIONS_SAVE_ROOT must be absolute paths
+MODEL_ROOT=<path to trained model>  # contains model.pth and config.yaml
+DETECTIONS_SAVE_ROOT=<path to save pre-computed detections>
+
+cd $VQ2D_ROOT
+
+# Extract per-frame bbox proposals and visual query similarity scores
+chmod +x ./scripts/extract_vq_detections.sh
+./scripts/extract_vq_detections.sh val $MODEL_ROOT $DETECTIONS_SAVE_ROOT
+./scripts/extract_vq_detections.sh test_unannotated $MODEL_ROOT $DETECTIONS_SAVE_ROOT
+```
+
+**Step (2)** Peak detection and bidirectional tracking.
+
+```
+./scripts/infer_vq.sh $MODEL_ROOT $DETECTIONS_SAVE_ROOT val 8 0.50 0.10
+./scripts/infer_vq.sh $MODEL_ROOT $DETECTIONS_SAVE_ROOT test_unannotated 8 0.50 0.10
+```
+
+**Notes:**
+* To reduce GPU / CPU usage, reduce 8 from step (2) based on your specific system.
+
+* To get VQ2D evaluation results:
     ```
-    cd $VQ2D_ROOT
-    python validate_challenge_predictions.py --test-unannotated-path <PATH TO vq_test_unannotated.json> --test-predictions-path <PATH to test_challenge_predictions.json>
+    python evaluate_vq.py --gt-file data/vq_val.json --pred-file <path to inference json>
     ```
+* To participate in the challenge, submit the inference json obtained for the test_unannotated split on evalai.
+
+## Pre-trained models and detection scores
+For reproducibility and conveneice,  we provide pre-trained models and corresponding detection scores for the [SiamRCNN baseline](https://arxiv.org/pdf/2110.07058.pdf) and [ImprovedBaselines model](https://github.com/facebookresearch/vq2d_cvpr). They can be downloaded using the ego4d CLI as follows:
+
+```
+python -m ego4d.cli.cli -y --output_directory /path/to/output/ --datasets vq2d_models vq2d_detections
```

Comment:
@ebyrne @devanshk - Is it possible to update the ego4d api datasets with newer files now? I'll have additional vq2d_models to release and pre-computed detections for these models in vq2d_detections.